### ISSUE TITLE
More better way for logging regexp-result / input history 

### ISSUE THREAD

**Issue by inafact** [p-code-magazine/r3pl#2]
random regexp でコードを展開した際にヒストリーとして何を格納するかで迷っている

- 現状では展開して実行した結果
- 展開前の正規表現の入力を保持しておいた方が面白いかもしれない？

**Comment by hysysk** [comment]
random regexpした後に確定状態にならず、入力中を保持してそのまま変更できるといいなと思いました。
その上で展開前の状態に戻せるのはありがたい気がします。

**Comment by inafact** [comment]
なるほど。これは実際には入力フローの中での振る舞い、ということですかね。

Ctrl+Spaceで展開→Enterで（実行せずに）展開結果を編集できますが、例えばその後Escで展開前の入力状態に戻れる、とか。

**Comment by hemokosa** [comment]
最初の話に戻るのですが、確かに展開前の正規表現のヒストリー機能があると、（僕としては）嬉しい感じです。キーバインディングで両方（展開前と展開後）できるようにするのは難しいでしょうか？

**Comment by inafact** [comment]
> 最初の話に戻るのですが、確かに展開前の正規表現のヒストリー機能があると、（僕としては）嬉しい感じです。キーバインディングで両方（展開前と展開後）できるようにするのは難しいでしょうか？

そうですね。コンセプトとしてどういう部分を押し出すかに拠るのかもという印象もありますが、僕も個人的には現状の実行結果のみをヒストリーに残していくより、正規表現自体をヒストリーに残していく方（あるいは提案してもらっているように両方残して、選択できるようにする）が良さそうかなという気がしています。

ヒストリー機能といいつつ、ある種未決定の状態を記録するみたいなことは、コードポエトリーであるとかライブコーディングの文脈にうまく繋がりそうな。

機能自体の追加は入力フロー・ヒストリー両方共可能だと思いますが、 #3 でも触れてるように一応モバイルデバイスでも使えるようにしたいというのもあるので、UI・UX的な部分をちょっと先に整理してみます。

**Comment by hemokosa** [comment]
未決定の状況を残すことは、ライブコーディングにおける可能世界を考える、という意味でも面白いかと。量子論的に、観測（実行）
）前の（未確定）状態と実行後の確定状態がパラに存在する、というか。
